src/ScaffoldInstallerPlugin.php

```diff
@@ -468,10 +468,26 @@ function($key) {
         // Add Redis service.
         if (file_exists('./.ddev/docker-compose.redis.yaml')) {
             $redisConfig = Yaml::parseFile('.ddev/docker-compose.redis.yaml');
-            $redisImage = explode(':',
-                $redisConfig['services']['redis']['image']);
+            $imageRaw = $redisConfig['services']['redis']['image'] ?? '';
+
+            // Normalize image from possible environment syntax.
+            if (preg_match('/^\$\{[^:}]+:-(.+)\}$/', $imageRaw, $matches)) {
+                $image = $matches[1];
+            } elseif (preg_match('/^([^:]+):\$\{[^:}]+:-(.+)\}$/', $imageRaw, $matches)) {
+                $image = "{$matches[1]}:{$matches[2]}";
+            } else {
+                $image = $imageRaw;
+            }
+
+            // Extract the version/tag from the image string.
+            if (preg_match('/:([a-zA-Z0-9._-]+)$/', $image, $versionMatch)) {
+                $version = $versionMatch[1];
+            } else {
+                $version = 'unknown';
```

Comment:
@deviantintegral Should we return unknown or have a fallback Redis version?

Reply:
Hm, I think this is tricky because if we hit this case we _know_ ddev start will fail. Why don't we throw a `\RuntimeException` instead, since we know a developer will have to intervene?